File: engine/lib/phx/script/meta/WorkerId.lua
---@meta

-- Types of workers.
-- Can be extended on the Lua side.
---@class WorkerId
---@field Echo integer Example worker that replicates input value into the output
---@field EngineWorkersCount integer Specifies number of engine worker types
WorkerId = {
    -- Example worker that replicates input value into the output
    Echo = 0,
    -- Specifies number of engine worker types
    EngineWorkersCount = 1,
}

File: engine/lib/phx/src/engine/task_queue/worker_id.rs
pub type WorkerIndex = u16;

/// Types of workers.
/// Can be extended on the Lua side.
#[luajit_ffi_gen::luajit_ffi(repr = "u16")]
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
pub enum WorkerId {
    /// Example worker that replicates input value into the output
    Echo,

    /// Specifies number of engine worker types
    EngineWorkersCount, // !!! SHOULD BE THE LAST ENUM VARIANT !!!
}

impl WorkerId {
    pub fn from_worker_id(id: WorkerIndex) -> Option<Self> {
        if id == Self::Echo as WorkerIndex {
            Some(Self::Echo)
        } else {
            None
        }
    }
}
